### PR TITLE
Add Teo TO8D emulator

### PR DESCRIPTION
**Pull request by hadess**
The screenshots might be a bit small...

### CODE REVIEW

File: net.sourceforge.Teo.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                "install -m0755 teo /app/bin/teo",
+                "install -m0644 -D misc/pack/repo/linux/debian/teo/usr/share/pixmaps/net.sourceforge.Teo.png /app/share/pixmaps/net.sourceforge.Teo.png",
+                "install -m0644 -D misc/pack/repo/linux/debian/teo/usr/share/applications/net.sourceforge.Teo.desktop /app/share/applications/net.sourceforge.Teo.desktop",
+                "sed -i 's,/usr/games/,/app/bin/,' /app/share/applications/net.sourceforge.Teo.desktop",
+                "install -m0644 -D misc/pack/repo/linux/debian/teo/usr/share/metainfo/net.sourceforge.Teo.appdata.xml /app/share/metainfo/net.sourceforge.Teo.appdata.xml"
+            ],
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "hg::http://hg.code.sf.net/p/teoemulator/code"
```
Comment (nedrichards) [inline net.sourceforge.Teo.json]: this needs to be a reproducible source. so a git tag+commit or a release archive to be available on flathub.

**Comment by hadess** [comment]
Done, the git commit IDs look stable even after being munged through git-hg.

I plan on replacing the screenshots once this Flatpak has its own module.

AppStream doesn't have any correct option for the translation method used for this software (which also runs under DOS and Windows), where the language shown is either French (when in a French locale), or English for anything else.

**Comment by nedrichards** [comment]
Ah, interesting.

**Comment by nedrichards** [comment]
bot, build net.sourceforge.Teo

**Comment by nedrichards** [comment]
The build failed: https://flathub.org/builds/#/builders/2/builds/2417
Looks like the cool workaround didn't work.

**Comment by hadess** [comment]
Of course, because flatpak-builder in this configuration will download everything first, then do the sequential build steps, and not download, build, download, build, etc.

I'll have to try and find a work-around for this, maybe a git mirror, or adding hg support to flatpak-builder (but then it'd need to be in the SDK somehow). If anyone has ideas...

**Comment by Alexander-Wilms** [comment]
Is there any reason not to use the archive at https://sourceforge.net/projects/teoemulator/files/linux/1.8.4/teo-1.8.4-src.tar.gz ?

**Comment by barthalion** [comment]
It needlessly increases the size of git repo with no way out in the future.

**Comment by Alexander-Wilms** [comment]
AFAIK, icons should live in `/app/share/icons/hicolor/{64x64,scalable,etc}/apps`.

**Comment by nedrichards** [comment]
Yeah, pixmaps isn't sufficient anymore.

**Comment by hadess** [comment]
Why is that now a requirement? Flatpak used to export those files just fine. And when the application has a single icon...

**Comment by Alexander-Wilms** [comment]
https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html mentions that PREFIX/share/icons is preferred and only mentions /usr/share/pixmaps, (not even PREFIX/share/pixmaps) as fallback.

**Comment by hadess** [comment]
Was removed in:
https://github.com/flatpak/flatpak/commit/d8ac76d467e742e896e86f419cd1837fa5542852

I can used /apps/pixmaps/icons instead. I'll do that once I've worked on the flatpak-builder patch some more.

**Comment by nedrichards** [comment]
bot, build net.sourceforge.Teo

**Comment by hadess** [comment]
Failed with:
```
*** Please tell me who you are.
Run
  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"
to set your account's default identity.
Omit --global to set the identity only in this repository.
fatal: empty ident name (for <buildbot@cncf-arm-1.flathub.org>) not allowed
```

There's a bug in the git-init patch I made for flatpak-builder. Or we could add info for git-commit to work in Flathub, not sure.

**Comment by hadess** [comment]
I've worked around the `git-init` problem in this last version, hopefully, by using `REAL_NAME` and `EMAIL_ADDRESS` envvars.

**Comment by nedrichards** [comment]
bot, build net.sourceforge.Teo

**Comment by nedrichards** [comment]
boooooooo https://flathub.org/builds/#/builders/2/builds/2833

**Comment by hadess** [comment]
> boooooooo https://flathub.org/builds/#/builders/2/builds/2833

Yep, the `build-options` aren't used when preparing the sources :/

I've updated the flatpak-builder patch and fully tested it now, so we'll just have to wait for it before merging this. If someone has other ideas to work-around the problem...

**Comment by nedrichards** [comment]
Sounds like we might have to wait until 1.0.1 then (plus rollout time).

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/net.sourceforge.Teo

**Pull request closed without merging**